### PR TITLE
upstream update 0.1.0

### PR DESCRIPTION
**Pull request by mkobetic**
Bring the latest upstream changes into our fork. This update doesn't look too bad, shouldn't have much impact.

Relevant changes

* new options
  --persist-messages
  --persist-peers
  --nat  (method to use for determining public address: any, none)
  --version (print waku version)
  --swap (enable swap)

* renamed options
  --store-days => --store-seconds

* persistence/store/DBStore.Query - respects orderDirection now

* node

   WakuNode now has enode.LocalNode with significant changes to address change and ENR handling.
   Seems to be related to https://github.com/status-im/go-waku/pull/252 (js-waku/nwaku interop)

### CODE REVIEW

File: waku/v2/protocol/filter/filter_subscribers.go [hunk 1]
```diff
@@ -15,6 +15,13 @@ type Subscriber struct {
 }
 
 func (sub Subscriber) HasContentTopic(topic string) bool {
+<<<<<<< HEAD
```
Comment (neekolas) [inline waku/v2/protocol/filter/filter_subscribers.go]: Looks like we have some merge conflicts in here

File: waku/v2/rpc/codec.go [hunk 2]
```diff
@@ -1,59 +1,105 @@
 package rpc
```
Comment (neekolas) [inline waku/v2/rpc/codec.go:1]: Is this whole JSON RPC thing new? I don't remember it being there.

File: waku/v2/protocol/filter/filter_subscribers.go (hunk 1 above)
Comment (mkobetic) [reply waku/v2/protocol/filter/filter_subscribers.go]: yup, fixed. I keep forgetting to save the file, after I resolve the conflicts 🤦

**Review by neekolas — APPROVED** [review]

**Comment by mkobetic** [comment]
I tried to rerun the tests a few times, the Test5000 is just not succeeding on the branch, I'll merge and hope that it will pass with our tweaks on master.

**Pull request merged**